doc/tutorial/single_cell_bluepyopt.rst

```diff
@@ -0,0 +1,288 @@
+
+To perform axon replacement, we first have to find the (unique) root segment of this region in the morphology. We can then split the segment tree at the axon root into two separate segment trees - a pruned one and one for the original axon.
+
+.. code-block:: python
+
+   # get unique axon root (tag for axon is 2)
+   axon_root = morpho.segment_tree.tag_roots(2)[0]
+
+   # prune morphology at axon root
+   pruned_st, axon_st = morpho.segment_tree.split_at(axon_root)
+
+Then we can determine the parent of the axon root to join the replacement onto, which we load from ACC. Alternatively, the user can build/supply a custom replacement here.
+
+.. code-block:: python
+
+   # parent segment to join onto
+   axon_parent = morpho.segment_tree.parents[axon_root]
+
+   # load replacement morphology
+   axon_replacement = arbor.load_component(
+      cell_json['morphology']['replace_axon']).component
+   axon_replacement_st = replacement.to_segment_tree()
+
+   # pruned segment tree with axon replacement
+   joined_st = pruned_st.join_at(
+      axon_parent, axon_replacement_st)
+
+   # final morphology
+   morpho_axon_replaced = arbor.morphology(joined_st)
+
+The steps presented here are included when loading an exported model with the above ``read_acc`` function from BluePyOpt.
+
+
+Define locations in the GUI
+---------------------------
+
+BluePyOpt makes use of the Neuron simulator's features to define custom locations for stimuli and voltage probes that can be challenging to translate to Arbor. As an example, the layer-5 pyramidal cell optimization problem has multiple voltage probes located distally on the apical dendrite at a fixed distance from the soma as described in `config/protocols.json <https://github.com/BlueBrain/BluePyOpt/blob/master/examples/l5pc/config/protocols.json>`_.
+
+To describe these locations accurately in Arbor, we will use the Arbor GUI. For this purpose, we create an :class:`arbor.cable_cell` and export it to ACC so that it can be loaded in the GUI,
+
+.. code-block:: python
+
+   cell = arbor.cable_cell(morpho, labels, decor)
+   arbor.write_component(cell, 'path/to/l5pc_cable_cell.acc')
+
+After loading this in file in the GUI with ``File > Cable cell > Load``, we can click the ``Locations`` and ``Cell`` tab to obtain the following overview.
+
+.. figure:: single_cell_bluepyopt_gui_l5pc_overview.png
+    :width: 800
+    :align: center
```

Comment:
A short caption on this figure and the others would be great. In subsequent figures, usually something specific is meant to be shown; it would be good to restate this in the caption.